test/lint_simp_var_head.lean

```diff
@@ -22,3 +22,15 @@ decl ← get_decl ``const_zero_eq_zero,
 res ← linter.simp_var_head.test decl,
 -- linter complains
 guard $ res.is_some
+
+
+
+-- However injectivity lemmas can still be marked simp,
+-- even though injective is reducible and unfolds to a bad simp lemma:
```

Comment:
What do you mean by "bad" here?

Reply:
Bad in the sense of the `simp_var_head` linter (which this file tests).  If you unfold `injective f`, then you get `∀ x y, f x = f y → x = y` where the left-hand side `x` is a variable.  This is bad because the simplifier code doesn't support this, and the linter makes sure we don't mark it as a simp lemma.